allennlp/training/model.py

```diff
@@ -0,0 +1,87 @@
+from typing import Any, List, Dict, Union
+import torch
+
+
+class Model(torch.nn.Module):
+    """
+    This abstract class represents a model to be trained. Rather than relying on the
+    raw torch Module, we provide a slightly different API.
+    """
+
+    def forward(self, inputs: Dict[str, torch.Tensor], compute_loss: bool = False):
```

Comment:
I think it makes more sense to separate loss computation from `forward`.  You already have `compute_loss` as its own function - why also include it in `forward`?  Is it because the targets are included in `inputs`?  We can separate that out somehow in the data, and that'd probably be cleaner than trying to handle it like this.